src/scripts/main.js

```diff
@@ -2,10 +2,20 @@
 
 function waitFor(element, eventName) {
```

Comment:
in this function you have to add event listener to element with proper eventName and do resolve of promise on it.